### PR TITLE
64 remove not redacted and check duplication of primary redaction

### PR DESCRIPTION
**Pull request by VT-AOE-DMAD-Drew-Bennett**

### CODE REVIEW

File: packages/suppression_check.py [hunk 1]
```diff
@@ -246,53 +251,72 @@ def redact_threshold(self):
...
+                        df_log_na.loc[condition, 'RedactBreakdown'] += ', Sum of values less than threshold'
         
-    #     # Update the 'Redact' column with a specific message for rows where 'MinimumValue' is not null
-    #     self.df_log.loc[mask, 'Redact'] = 'Secondary Suppression'
-    #     #self.df_log.loc[mask, 'Redact'] = 'Sum of minimum threshold redact needed secondary suppression'
+        else:
+            for sensitive_combination in self.sensitive_combinations:
+                list_combination = list(sensitive_combination)
+                if list_combination != self.sensitive_columns:
+                    string_combination = ''.join(list_combination)
```
Comment (sethltaylor) [inline packages/suppression_check.py:296]: My IDE shows string_combination as unused.

File: packages/suppression_check.py [hunk 2]
```diff
@@ -246,53 +251,72 @@ def redact_threshold(self):
...
+        for sensitive_column in self.sensitive_columns:
+            df_log_na[sensitive_column].fillna(temp_value, inplace=True)  
         
-    #     self.df_log.loc[mask, 'RedactBinary'] = 1
+        # Grouping by Organization and counting StudentCount, then filtering groups with a single record 
+        if self.organization_columns[0] is not None: 
+            for sensitive_combination in self.sensitive_combinations:
+                list_combination = list(sensitive_combination)
+                if list_combination != self.sensitive_columns:
+                    string_combination = ''.join(list_combination)
```
Comment (sethltaylor) [inline packages/suppression_check.py:275]: IDE shows string_combination as unused.

File: packages/suppression_check.py [hunk 3]
```diff
@@ -246,53 +251,72 @@ def redact_threshold(self):
...
+                    df_redact_less = df_log_na[df_log_na['RedactBinary'] == 1]
+                    df_redact_less.loc[:, 'Redacted'] = 1
+                    df_count = df_redact_less.groupby(['Grouping'] + self.organization_columns + list_combination)['Redacted'].sum().reset_index()
```
Comment (sethltaylor) [inline packages/suppression_check.py:278]: df_count is marked as unused as well

File: packages/suppression_check.py [hunk 4]
```diff
@@ -246,53 +251,72 @@ def redact_threshold(self):
...
+                    df_summed_less_than = df_summed_value[df_summed_value[self.frequency] <= self.minimum_threshold]
```
Comment (sethltaylor) [inline packages/suppression_check.py:279]: df_summed_value is referenced before it's assigned. Was it supposed to be assigned based on df_count instead?

File: packages/suppression_check.py [hunk 5]
```diff
@@ -246,53 +251,72 @@ def redact_threshold(self):
...
-    #     self.df_log.loc[mask, 'Redact'] = 'Secondary Suppression'
-    #     #self.df_log.loc[mask, 'Redact'] = 'Sum of minimum threshold redact needed secondary suppression'
+        else:
+            for sensitive_combination in self.sensitive_combinations:
+                list_combination = list(sensitive_combination)
+                if list_combination != self.sensitive_columns:
+                    string_combination = ''.join(list_combination)
+                    df_redact_less = df_log_na[df_log_na['RedactBinary'] == 1]
+                    df_redact_less.loc[:, 'Redacted'] = 1
+                    df_summed_value = df_redact_less.groupby(['Grouping'] + list_combination, dropna=False)[self.frequency].sum().reset_index()
```
Comment (sethltaylor) [inline packages/suppression_check.py:299]: Looks like df_summed_value is assigned here, so the bug above will only show up when organization columns are specified.

**Review by sethltaylor — COMMENTED** [review]
Drew there are a couple unreferenced assignments and one bug that I think just requires changing a variable name.

File: packages/suppression_check.py (hunk 2 above)
Comment (VT-AOE-DMAD-Drew-Bennett) [reply packages/suppression_check.py:275]: It is being used for the loop of sensitive columns. Also, I ran a few tests and it works.

**Review by sethltaylor — APPROVED** [review]

**Pull request merged**